src/main/kotlin/com/launchdarkly/intellij/hover/HoverDocumentationProvider.kt

```diff
@@ -70,10 +72,11 @@ class HoverDocumentationProvider : AbstractDocumentationProvider() {
     }
 
     override fun generateDoc(element: PsiElement?, originalElement: PsiElement?): String? {
-        // TODO: if element is of type "empty token" then exit immediately.
-        // This can happen when a go project is opened in intellij i.e. when a project type
-        // is unsupported by the IDE flavor.
-        if (element == null || element == StandardPatterns.not(
+        // Immediately exit if there's no PSIElement. In other words, if the language is not supported
+        // by the IDE, then we can't show any docs for it.
+        // https://intellij-support.jetbrains.com/hc/en-us/community/posts/360008223759/comments/360001676819
+        val type = (element as? LeafPsiElement)?.elementType
+        if (element == null || element.language == Language.ANY || element.textLength > Utils.FLAG_KEY_MAX_LENGTH || type.toString() == "empty token" || element == StandardPatterns.not(
```

Comment:
If a language is not supported, it looks like the IDE labels it as "Language.ANY". Secondly in a truly cowboy style, it is possible that a language is supported but the node contents is the entire file, hence the second check for max length.

Reply:
I think its worth documenting each of these via comments, it'll be confusing for the next person who looks at this line of code (or future Yus who has forgotten about this)